stko/optimizers/xtb.py

```diff
@@ -496,3 +514,633 @@ def optimize(self, mol):
+                f'The chosen speed setting {speed_setting} will ',
+                f'override the chosen energy window {ewin}.'
+            )
+
+        self._ewin = ewin
+        self._speed_setting = speed_setting
+        self._keepdir = keepdir
+        self._num_cores = str(num_cores)
+        self._charge = str(charge)
+        self._unlimited_memory = unlimited_memory
+
+    def _is_complete(self, output_file, output_xyzs):
+        """
+        Check if CREST run has completed.
+
+        Parameters
+        ----------
+        output_file : :class:`str`
+            Name of CREST output file.
+
+        output_xyzs : :class:`str`
+            Name of CREST conformer and rotamer output files.
+            crest_best.xyx > Best conformer, exists throughout run.
+            crest_conformers.xyx > All conformers,
+                exists throughout run.
+            crest_rotamers.xyx > All rotamers, exists one run is done.
+
+        Returns
+        -------
+        :class:`bool`
+            Returns ``False`` if a negative frequency is present.
+
+        Raises
+        -------
+        :class:`CRESTNotStartedError`
+            If the CREST run failed to start.
+
+        :class:`CRESTNotCompletedError`
+            If the CREST run failed to complete.
+
+        """
+
+        if not os.path.exists(output_file):
+            # No simulation has been run.
+            raise CRESTNotStartedError('CREST run did not start.')
+
+        elif any(not os.path.exists(i) for i in output_xyzs):
+            # Best conformer was not output.
+            raise CRESTNotCompletedError('CREST run did not complete.')
+
+        return True
+
+    def _run_crest(self, xyz, out_file):
+        """
+        Run CREST along side GFN-xTB.
+
+        Parameters
+        ----------
+        xyz : :class:`str`
+            The name of the input structure ``.xyz`` file.
+
+        out_file : :class:`str`
+            The name of output file with xTB results.
+
+        Returns
+        -------
+        None : :class:`NoneType`
+
+        """
+
+        # Modify the memory limit.
+        if self._unlimited_memory:
+            memory = 'ulimit -s unlimited ;'
+        else:
+            memory = ''
+
+        # Set optimization level and type.
+        optimization = f'-opt {self._opt_level}'
+        keepdirs = '-keepdir' if self._keepdir is True else ''
+        speed_settings = (
+            f'-{self._speed_setting}'
+            if self._speed_setting is not None else ''
+        )
+
+        cmd = (
+            f'{memory} {self._crest_path} {xyz} '
+            f'-xnam {self._xtb_path} -nozs '
+            f'-ewin {self._ewin} '
+            f'-chrg {self._charge} '
+            f'-gff {speed_settings} '
+            f'{optimization} -T {self._num_cores} {keepdirs}'
+        )
+
+        with open(out_file, 'w') as f:
+            # Note that sp.call will hold the program until completion
+            # of the calculation.
+            sp.call(
```

Comment:
This can be changed to run in future versions I assume?

Reply:
Don't know what you mean?

Reply:
There are no depracation warnings for sp.call.